sql/statistics.go

```diff
@@ -174,11 +179,44 @@ type StatisticsTable interface {
 	Table
 	// DataLength returns the length of the data file (varies by engine).
 	DataLength(ctx *Context) (uint64, error)
-	// AnalyzeTable is a hook to update any cached or persisted statistics for this table.
-	// It may be triggered by an analyze table statement, or automatically when the engine decides it is necessary.
-	// Integrators can ignore this hook and implement their own method of keeping statistics up to date, at the
-	// cost of potentially stale statistics.
-	AnalyzeTable(ctx *Context) error
-	// Statistics returns the statistics for this table
-	Statistics(ctx *Context) (TableStatistics, error)
+	// RowCount returns the statistics for this table
+	RowCount(ctx *Context) (uint64, error)
+}
+
+// CatalogTable is a Table that depends on a Catalog.
+type CatalogTable interface {
+	Table
+
+	// AssignCatalog assigns a Catalog to the table.
+	AssignCatalog(cat Catalog) Table
+}
+
+type StatsReader interface {
```

Comment:
Document this interface, say it's for internal use only

Reply:
the methods are already in Dolt preceding this PR, i just shifted around the interface clusters. Stats implementing updater is unique and internal only.

Reply:
granted its not used unless someone calls `ANALYZE ...`